src/backend/egl/fence.rs

```diff
@@ -0,0 +1,163 @@
+//! EGL fence related structs
+
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, OwnedFd};
+use std::sync::Arc;
+use std::time::Duration;
+
+use crate::backend::egl::{ffi, wrap_egl_call, EGLDisplay, EGLDisplayHandle, Error};
+
+#[derive(Debug)]
+struct InnerEGLFence {
+    display_handle: Arc<EGLDisplayHandle>,
+    handle: ffi::egl::types::EGLSync,
+    native: bool,
+}
+unsafe impl Send for InnerEGLFence {}
+unsafe impl Sync for InnerEGLFence {}
+
+impl Drop for InnerEGLFence {
+    fn drop(&mut self) {
+        unsafe {
+            // Not much we can do about when this fails
+            ffi::egl::DestroySync(**self.display_handle, self.handle);
+        }
+    }
+}
+
+/// A EGL fence
+#[derive(Debug, Clone)]
+pub struct EGLFence(Arc<InnerEGLFence>);
+
+impl EGLFence {
+    /// Returns whether the display supports creating fences from native fences
+    pub fn supports_from_native(display: &EGLDisplay) -> bool {
+        display.supports_native_fences
+    }
+
+    /// Create a fence from a native fence
+    pub fn from_native(display: &EGLDisplay, native: OwnedFd) -> Result<Self, Error> {
```

Comment:
Would it be more accurate to call this importing instead? Vulkan uses the term import for this type of native sync fd to fence

Reply:
tbh I have no preference, both is fine for me. @Drakulix any preference?